frontend/src/app/pages/exercises/exercise/shared/exercise-statistics/hospital-patients-table/hospital-patients-table.component.ts

```diff
@@ -0,0 +1,82 @@
+import { Component } from '@angular/core';
+import type { Sort } from '@angular/material/sort';
+import { Store } from '@ngrx/store';
+import { statusNames } from 'digital-fuesim-manv-shared';
+import type { Observable } from 'rxjs';
+import { combineLatest, map, startWith, Subject } from 'rxjs';
+import type { AppState } from 'src/app/state/app.state';
+import {
+    selectHospitalPatients,
+    selectHospitals,
+} from 'src/app/state/exercise/exercise.selectors';
+
+@Component({
+    selector: 'app-hospital-patients-table',
+    templateUrl: './hospital-patients-table.component.html',
+    styleUrls: ['./hospital-patients-table.component.scss'],
+})
+export class HospitalPatientsTableComponent {
+    constructor(public readonly store: Store<AppState>) {}
+
+    public readonly sortEvent$ = new Subject<Sort>();
+
+    private readonly unsortedRows$: Observable<HospitalPatientsRow[]> =
+        combineLatest([
+            this.store.select(selectHospitalPatients),
+            this.store.select(selectHospitals),
+        ]).pipe(
+            map(([hospitalPatients, hospitals]) =>
+                Object.values(hospitalPatients).map((hospitalPatient) => {
+                    const hospital = Object.values(hospitals).find(
+                        (_hospital) =>
+                            _hospital.patientIds[hospitalPatient.patientId]
+                    );
+                    const row: HospitalPatientsRow = {
+                        hospitalName:
+                            hospital?.name ?? 'unbekanntes Krankenhaus',
```

Comment:
This should probably not be possible, as right now, when a hospital is deleted, every patient in it gets also deleted.
Can keep it in, but maybe with a comment that this is probably not possible or so.
Or we change `removeHospital` to not delete patients (which makes them undeletable, except this would be a function added to this table).

Reply:
In most cases, creation and deletion of hospitals will be BEFORE an exercise is running and never during an exercise, so the question of what happens to patients transported to a deleted hospital is an edge case.

However, I would prefert it if the patients remain in the list with a text "deleted hospital".